index.js

```diff
@@ -363,28 +362,28 @@ function tokensToRegExp (tokens, options) {
  * placeholder key descriptions. For example, using `/user/:id`, `keys` will
  * contain `[{ name: 'id', delimiter: '/', optional: false, repeat: false }]`.
  *
- * @param  {(String|RegExp|Array)} path
- * @param  {Array}                 [keys]
- * @param  {Object}                [options]
- * @return {RegExp}
+ * @param  {(string|RegExp|Array)} path
+ * @param  {(Array|Object)=}       keys
+ * @param  {Object=}               options
+ * @return {!RegExp}
  */
 function pathToRegexp (path, keys, options) {
   keys = keys || []
 
   if (!isarray(keys)) {
-    options = keys
+    options = /** @type {!Object} */ (keys)
     keys = []
   } else if (!options) {
     options = {}
   }
 
   if (path instanceof RegExp) {
-    return regexpToRegexp(path, keys, options)
+    return regexpToRegexp(path, /** @type {!Array} */ (keys))
```

Comment:
The compiler warned that `regexpToRegexp` was being called with more arguments than it expected. While this isn't actually cause errors, more often that you would expect it indicates that the caller assumed a behavior that wasn't actually supported. In this case, passing the options argument does nothing. This was the only actual code change in the PR.

The other annotation is a type cast to tighten the type of the parameter. If the if test preceding this was of the form `keys instanceof Array`, the compiler would do this for us, but here it can't tell that the `isarray` method is used to tighten the type.